lib/core/Observer.ts

```diff
@@ -114,14 +113,27 @@ export default class Observer {
           void this.processTransaction(transaction, awaitingTransaction);
         }
 
+        // NOTE: Blockchain reorg has happened for sure only if `invalidTransactionNumberOrTimeHash` AND
+        // latest transaction time is less or equal to blockchain service time.
+        // This check will prevent Core from reverting transactions if/when blockchain service is reinitializing its data itself.
+        let blockReorganizationDetected = false;
```

Comment:
The only reliable way I know for detecting rollbacks is to look for the least-common ancestor from the stored headers and what bitcoin has, see [this](https://github.com/X9Developers/block-explorer/blob/develop/server/app/com/xsn/explorer/services/synchronizer/LedgerSynchronizationStatusService.scala#L42) for details.

Reply:
You are right @AlexITC, this logic is handled in the blockchain (bitcoin) microservice.